roles/hetzner-network.nix

```diff
@@ -37,14 +24,13 @@ in {
     networking.dhcpcd.enable = false;
     systemd.network = {
       enable = true;
-      networks."eth0".extraConfig = ''
+      networks."ethernet".extraConfig = ''
         [Match]
-        Name = eth0
+        Type = ether
         [Network]
+        DHCP = ipv4
```

Comment:
Let's see if this fixes this issue. At least it is less code, which is also good.